### ISSUE TITLE
Failure on big files with S3 storage

### ISSUE THREAD

**Issue by mihalo** [janko/tus-ruby-server#14]
I'm trying to upload +100gb files to S3 storage but I'm getting error:

`FATAL -- : Aws::S3::Errors::InvalidArgument (Part number must be an integer between 1 and 10000, inclusive)`

Is there any workaround or fix for this error?

**Comment by janko** [comment]
The problem is that `tus-ruby-server` uses a fixed chunk size of 5MB, so for large files it exceeds the maximum number of multipart parts allowed by S3.

I started working this a while ago, basically adapting the calculating to be the same as [tusd does](https://github.com/tus/tusd/blob/25f64ee00a22b0a078e9bcffea3d0085389b8d4d/s3store/s3store.go#L683-L725), but I haven't finished it. I will try to address this as soon as I find free time.

For now you should be able to work around it by increasing the `Tus::Storage::S3::MIN_PART_SIZE` constant to a sufficiently large value (such that `MIN_PART_SIZE * 10_000 > maximum total filesize`). For example, for a 100GB file 15MB would suffice:

```rb
Tus::Storage::S3::MIN_PART_SIZE = 15 * 1024 * 1024
```

You'll unfortunately get the warning that you're changing the constant, but hopefully it should work.

**Comment by mihalo** [comment]
Thank you!

I made a workaround for this by passing uploadChunkSize in metadata from `tus-js-client` as I can calculate optimal chunk size there and use it in client and server...

Maybe not the cleanest solution but at least it works :)

path_file function looks something like this:

```
def patch_file(uid, input, info = {})
	tus_info = Tus::Info.new(info)
	min_size = MIN_PART_SIZE
        
	if tus_info.metadata["uploadChunkSize"]
		min_size = tus_info.metadata["uploadChunkSize"].to_i
	end
	.....
end
```

**Comment by janko** [comment]
I just released 2.3.0 which should support uploads of any size (without any additional configuration).
